tasks/main.yml

```diff
@@ -2,65 +2,28 @@
 - name: Set Rsyslog facts then include rsyslog role
   block:
 
-    - name: Set rsyslog_output_elasticsearch for the omelasticsearch output
-      set_fact:
-        rsyslog_output_elasticsearch: "{{ logging_outputs |
-          selectattr('name', 'defined') |
-          selectattr('type', 'defined') |
-          selectattr('type', 'match', '^elasticsearch$') |
-          selectattr('server_host', 'defined') | list }}"
-
-    - name: Set rsyslog_output_files for the omfile outputs
-      set_fact:
-        rsyslog_output_files: "{{ logging_outputs |
-          selectattr('name', 'defined') |
-          selectattr('type', 'defined') |
-          selectattr('type', 'match', '^files$') | list }}"
-
-    - name: "Set files output if rsyslog_output_files is empty
+    - name: "Set files output if files output is not defined
       and logging_inputs is not empty"
       set_fact:
-        rsyslog_output_files: [ {"name": "default_files", "type": "files"} ]
+        _rsyslog_output_files: [ {"name": "default_files", "type": "files"} ]
       when:
-        - rsyslog_output_files | d([]) | length == 0
-        - logging_inputs | d([])
-
-    - name: Set rsyslog_output_forwards for the omfwd output
-      set_fact:
-        rsyslog_output_forwards: "{{ logging_outputs |
-          selectattr('name', 'defined') |
+        - logging_outputs | selectattr('name', 'defined') |
           selectattr('type', 'defined') |
-          selectattr('target', 'defined') |
-          selectattr('type', 'match', '^forwards$') | list }}"
+          selectattr('type', 'match', '^files$') |
+          list | d([]) | length == 0
+        - logging_inputs | d([])
 
-    - name: "Set rsyslog_output_remote_files for the omfwd output
-      for the remote input"
+    - name: Set rsyslog_outputs
       set_fact:
-        rsyslog_output_remote_files: "{{ logging_outputs |
-          selectattr('name', 'defined') | selectattr('type', 'defined') |
-          selectattr('type', 'match', '^remote_files$') | list }}"
+        rsyslog_outputs: "{{ logging_outputs | selectattr('name', 'defined') |
+          selectattr('type', 'defined') }} +
+          {{ _rsyslog_output_files | d([]) }}"
```

Comment:
Not a correction but a question, why some variables you introduced are prefixed with one underscore, not two underscores as usual? These are `_rsyslog_output_files`, `_input`, `_output`.

Reply:
Ah, I did not think about it much... :p Thanks for pointing it out, @spetrosi. I'm going to replace them with two underscores.

Reply:
@spetrosi, fixed. I also replaced `_input` and `_output` with `__rsyslog_input` and `__rsyslog_output` to make them less ambiguous. Thanks, again!

Reply:
Thank you for doing this, that's much clearer. I would wait for the CI results now.